src/TheFox/Smtp/Server.php

```diff
@@ -252,4 +262,28 @@ public function mailNew($from, $rcpt, $mail){
 		$this->eventExecute(Event::TRIGGER_MAIL_NEW, array($from, $rcpt, $mail));
 	}
 	
+	/**
+	 * Execute authentication events
+	 * 
+	 * All authentication events must return true for authentication to be successful
+	 * 
+	 */
+	public function authenticateUser($method, $credentials = array()){
+		$authenticated = false;
+		$args = array($method, $credentials);
+		
+		foreach($this->events as $eventId => $event){
+			if($event->getTrigger() == Event::TRIGGER_AUTH_ATTEMPT){
+				if($event->execute($args)){
+					$authenticated = true;
+				}
+				else{
+					$authenticated = false;
```

Comment:
Let's assume you set more then one `Event::TRIGGER_AUTH_ATTEMPT ` events. The `else` let the `authenticateUser()` end as `false` even if one of `Event::TRIGGER_AUTH_ATTEMPT ` events was `true`.